controllers/limitador_controller.go

```diff
@@ -278,3 +291,22 @@ func mutateLimitadorDeployment(existingObj, desiredObj client.Object) (bool, err
 
 	return updated, nil
 }
+
+func getStorageConfigSecret(ctx context.Context, client client.Client, secretRef *v1.ObjectReference) (*v1.Secret, error) {
+	storageConfigSecret := &v1.Secret{}
+	if err := client.Get(
+		ctx,
+		types.NamespacedName{
+			Name:      secretRef.Name,
+			Namespace: secretRef.Namespace,
```

Comment:
The namespace I would leave optional. So if not set, it takes the value from the namespace of the limitador's object